src/frontend/workers/folderWatcher.worker.ts

```diff
@@ -23,22 +24,47 @@ export class FolderWatcherWorker {
   /** Returns all supported image files in the given directly, and callbacks for new or removed files */
   async watch(directory: string) {
     this.isCancelled = false;
-    // Only watch for images files, something like /^.*\.(?!jpg$|png$)[^.]+$/i -> https://regexr.com/5pvbu
-    const imageExtensionIgnorePattern = new RegExp(
-      `^.*\\.(?!${IMG_EXTENSIONS.join('$|')}$)[^.]+$`,
-      'i',
-    );
 
-    // Watch for changes
+    // Replace backslash with forward slash, recommendeded by chokidar
+    // See docs for the .watch method: https://github.com/paulmillr/chokidar#api
+    directory = directory.replace(/\\/g, '/');
+
+    // Watch for files being added/changed/removed:
+    // Usually you'd include a glob in the watch argument, e.g. `directory/**/.{jpg|png|...}`, but we cannot use globs unfortunately (see disableGlobbing)
+    // Instead, we ignore everything but image files in the `ignored` option
     this.watcher = chokidar.watch(directory, {
-      disableGlobbing: true, // needed in order to support folders with brackets, quotes, etc.
-      depth: RECURSIVE_DIR_WATCH_DEPTH,
-      ignored: [
-        // Ignore dot files. Also dot folders?
-        /(^|[\/\\])\../,
-        // And non-image files
-        imageExtensionIgnorePattern,
-      ],
+      disableGlobbing: true, // needed in order to support directories with brackets, quotes, asterisks, etc.
+      alwaysStat: true, // we need stats anyways during importing
+      depth: RECURSIVE_DIR_WATCH_DEPTH, // not really needed: added as a safety measure for infinite recursion between symbolic links
+      ignored: (path: string, stats?: Stats) => {
+        // We used to set `ignored` with regex patterns, but ran into problem with directories that include dots:
+        // - we ignore everything but image files, but:
+        // - chokidar also matches entire directories: if those contain a dot, it will be ignored too, since they don't end with an image extension
+        // So now we use a callback function that also provides `stats` through which we can detect whether the path is a file or a directory
+
+        const basename = SysPath.basename(path);
+        const ext = SysPath.extname(path).toLowerCase().split('.')[1];
+
+        // If the path doesn't have an extension: it's likely a directory: don't ignore
+        // but if it's a .dot directory, it should be ignored: will be checked later
+        // In the unlikely situation it's a file, we'll filter it out later in the .on('add', ...)
+        if (!ext && !basename.startsWith('.')) return false;
```

Comment:
This does not ignores files with starting with dots. Should we filter them? Most people do not start images with dots.

Reply:
mhh, yeah, good point!

Reply:
Another option is to check if it is an image in the first place - the so called magic number test.